src/app/common/components/PipelineSummary.tsx

```diff
@@ -1,19 +1,93 @@
 import * as React from 'react';
+import { Flex, FlexItem, Text } from '@patternfly/react-core';
+import {
+  ResourcesAlmostEmptyIcon,
+  ResourcesAlmostFullIcon,
+  ResourcesFullIcon,
+} from '@patternfly/react-icons';
 
-import { Text } from '@patternfly/react-core';
+import {
+  global_danger_color_100 as dangerColor,
+  global_disabled_color_200 as disabledColor,
+  global_info_color_100 as infoColor,
+  global_success_color_100 as successColor,
+} from '@patternfly/react-tokens';
+
+import { IVMStatus } from '@app/queries/types';
+import './PipelineSummary.css';
 
 interface IPipelineSummaryProps {
-  total: number;
-  current: number;
-  status: string;
+  status: IVMStatus;
 }
 
+const Dash = (isReached: boolean): JSX.Element => {
+  return (
+    <FlexItem alignSelf={{ default: 'alignSelfCenter' }}>
+      {isReached ? <div className="dash dashReached" /> : <div className="dash dashNotReached" />}
+    </FlexItem>
+  );
+};
+
 const PipelineSummary: React.FunctionComponent<IPipelineSummaryProps> = ({
-  total,
-  current,
   status,
 }: IPipelineSummaryProps) => {
-  return <Text>{status}</Text>;
+  let sum: React.ReactNode;
+
+  const chain = (Face, times, color) => {
```

Comment:
It doesn't matter much, but technically a function which takes params and returns JSX should be formatted as a component. So this could be `<Chain>` with Face, times and color as props. Either way, the args/props should have TS types.

I see that it's recursive.. I'm actually not sure if components are allowed to render recursively? I think so.

Reply:
@mturley, it should be definitely be `Chain`.

Well not sure about that rendering part either. Is Chain rendered every time? I'm understanding it as building a JSX.Element not rendering it, but maybe that's what it means.

Reply:
Actually Chain is a `React.ReactNode`.

Reply:
I've changed Chain to be a `JSX.Element`.

Reply:
@mturley, Well spotted, thank you!

Reply:
I think `React.ReactNode` was fine, I actually wasn't aware of the JSX.Element type. Not sure the difference there.

Reply:
> I'm understanding it as building a JSX.Element not rendering it, but maybe that's what it means.

As far as I understand, I think that is what it means. It's doing essentially the same thing any of our other small components are doing, the only weird part is how it is recursive, but I think that will still work fine (?)

Reply:
React.FC gives a functional programming approach, isn't?